### PR TITLE
Using real models in trip flow

### PR DESCRIPTION
**Pull request by AsafGartner**

### CODE REVIEW

File: app/src/main/java/com/iic/shopingo/ui/trip_flow/activities/ManageTripActivity.java [hunk 1]
```diff
@@ -32,24 +34,28 @@ public void onCreate(Bundle savedInstanceState) {
...
+              "Moshe",
+              Integer.toString(i),
+              "AVATAR",
+              "12345",
+              "13 Rothschild Ave.",
+              "Tel Aviv",
+              32.063146,
+              34.770706
+          ),
+          new com.iic.shopingo.dal.models.ShoppingList(items, 500),
```
Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/ui/trip_flow/activities/ManageTripActivity.java:56]: Just the `ShoppingList`

File: app/src/main/java/com/iic/shopingo/ui/trip_flow/activities/ManageTripActivity.java [hunk 2]
```diff
@@ -92,11 +98,11 @@ public int getCount() {
...
 
   @Override
-  public void onRequestAccepted(Request request) {
+  public void onRequestAccepted(IncomingRequest request) {
     ShoppingList sl = new ShoppingList();
-    sl.requesterName = request.name;
-    sl.phoneNumber = request.location.phone;
-    for (String itemTitle : request.items) {
+    sl.requesterName = request.getRequester().getFirstName();
+    sl.phoneNumber = request.getRequester().getPhoneNumber();
```
Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/ui/trip_flow/activities/ManageTripActivity.java:104]: **General Question:** What impact does repeating all these `getRequester().getX()` and `getShoppingList().getY()` have on performance?

**Comment by assafgelber** [comment]
Awesome possum

File: app/src/main/java/com/iic/shopingo/ui/trip_flow/activities/ManageTripActivity.java (hunk 2 above)
Comment (AsafGartner) [inline app/src/main/java/com/iic/shopingo/ui/trip_flow/activities/ManageTripActivity.java:104]: I imagine it heavily depends on Dalvik/ART.
Technically, it just needs to load a pointer, which should already be in cache for the 2nd line, and the runtime should JIT the whole thing to do just that.
What it actually does, I have no idea.
I imagine you'd want a local variable for readability if you have to do it more than a few times.

Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/ui/trip_flow/activities/ManageTripActivity.java:104]: :+1:

**Pull request merged**